examples/plot_ranking.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+=========================================
+Ranking
+=========================================
+"""
+
+from fairlearn.metrics import exposure, utility, proportional_exposure
+from fairlearn.metrics import MetricFrame
+
+# %%
+# This notebook shows how to apply functionalities of :mod:`fairlearn.metrics` to ranking problems.
+# We showcase the example "Fairly Allocating Economic Opportunity" from the paper
+# `"Fairness of Exposure in Ranking" <https://dl.acm.org/doi/10.1145/3219819.3220088>`_
+# by Singh and Joachims (2018).
+# The example demonstrates how small differences in item relevance can lead to large differences
+# in exposure. Differences in exposure can be harmful not only because you are ranking individuals
+# directly but individuals can also be indirect affected by a ranking, think about books, music or
+# products in online retail.
+#
+# We reproduce the example of the paper, which for simplicity reasons uses a binary gender category
+# as sensitive attribute. The metric however also works for multi-class sensitive attributes.
+#
+# Consider a web-service that connects employers ("users") to potential employees ("items").
+# The web-service uses a ranking-based system to present a set of 6 applicants of which 3 are male
+# and 3 are female. Male applicants have relevance of 0.80, 0.79, 0.78 respectively for the
+# employer, while female applicants have relevance of 0.77, 0.76, 0.75.
+# In this setting a relevance of 0.75 is defined as, 75% of all employers issuing the query
+# considered the applicant relevant.
+#
+# A simple way to rank the participants is in decreasing order of relevance. What does this mean
+# for the exposure between the two groups?
+#
+# NOTE: The used data should raise questions about
+# :ref:`construct validity <fairness_in_machine_learning.construct-validity>` , since we
+# should consider whether the "relevance" that is in the data is a good measurement of the actual
+# relevance. Hindsight bias is also a concern, since how can you know upfront if an applicant
+# will be a successful employee in the future? These concerns are important, but out of scope for
+# this use case example.
+
+
+ranking_pred = [1, 2, 3, 4, 5, 6]  # ranking
+gender = ['Male', 'Male', 'Male', 'Female', 'Female', 'Female']
```

Comment:
We might want to use 'man' / 'woman' instead of 'male' / 'female' across the example (see also https://github.com/fairlearn/fairlearn/issues/970)

Reply:
Did not know there was a discussion between man/male woman/female. Reading #970, I will also emphasize that participants are labelled man/woman.